orchestrate/src/vm/error.rs

```diff
@@ -37,9 +34,11 @@ pub enum VmError {
     NotAuthorized,
 }
 
+impl wasmi::core::HostError for VmError {}
+
 impl From<wasmi::Error> for VmError {
     fn from(e: wasmi::Error) -> Self {
-        Self::Interpreter(e)
+        Self::Interpreter(format!("{e}"))
```

Comment:
Why do we want to format it? Because of the dyn issue?

Reply:
because `wasmi::Error` doesn't implement `PartialEq`, hence I converted it to `String`

Reply:
How is the user catching the actual host error then?

Reply:
currently user can only print the error details but is it useful to get the specific error type in orchestrate?

Reply:
It's useful in any client, as you would like to check for the error happening, in orchestrate it's useful because we inspect the error when we expect an error. Is there any chance we create a new error type that we wrap? Such that it inspect and unwrap the trap error to the VM error automatically? (for the actual wasm error like GlobalError etc its fine to have them as a string, but the host error in Trap must be extracted)

Reply:
We now, catch the host errors and downcast them to `VmError`